src/transformers/models/vits/modeling_vits.py

```diff
@@ -0,0 +1,1413 @@
+        >>> outputs.audio.shape
+        torch.Size([1, 1, 33280])
+        ```
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if attention_mask is not None:
+            input_padding_mask = attention_mask.unsqueeze(-1).float()
+        else:
+            input_padding_mask = torch.ones_like(input_ids).unsqueeze(-1).float()
+
+        # Workaround for tokenizer: filter out padding tokens.
+        input_ids[input_ids >= self.config.vocab_size] = 0
+
+        hidden_states, means_prior, log_variances_prior = self.text_encoder(
+            input_ids, input_padding_mask, attention_mask
+        )
+
+        hidden_states = hidden_states.transpose(1, 2)
+        input_padding_mask = input_padding_mask.transpose(1, 2)
+
+        if self.config.num_speakers > 1:
+            if speaker_id is None:
+                raise ValueError("Expected speaker_id")
+            speaker_embeddings = self.embed_speaker(torch.tensor([speaker_id])).unsqueeze(-1)
+        else:
+            speaker_embeddings = None
+
+        if labels is not None:
+            raise NotImplementedError("Training of VITS is not supported yet.")
+
+        if self.config.use_stochastic_duration_prediction:
+            log_duration = self.duration_predictor(
+                hidden_states, input_padding_mask, speaker_embeddings, reverse=True, noise_scale=noise_scale_duration
+            )
+        else:
+            log_duration = self.duration_predictor(hidden_states, input_padding_mask, speaker_embeddings)
+
+        length_scale = 1.0 / speed
+        duration = torch.ceil(torch.exp(log_duration) * input_padding_mask * length_scale)
+        predicted_lengths = torch.clamp_min(torch.sum(duration, [1, 2]), 1).long()
+
+        # Create a padding mask for the output lengths of shape (batch, 1, max_output_length)
+        indices = torch.arange(predicted_lengths.max(), dtype=predicted_lengths.dtype, device=predicted_lengths.device)
+        output_padding_mask = indices.unsqueeze(0) < predicted_lengths.unsqueeze(1)
+        output_padding_mask = output_padding_mask.unsqueeze(1).to(input_padding_mask.dtype)
+
+        # Reconstruct an attention tensor of shape (batch, 1, out_length, in_length)
+        attn_mask = torch.unsqueeze(input_padding_mask, 2) * torch.unsqueeze(output_padding_mask, -1)
+        attn = self._generate_path(duration, attn_mask)
+
+        # Expand prior
+        means_prior = torch.matmul(attn.squeeze(1), means_prior).transpose(1, 2)
+        log_variances_prior = torch.matmul(attn.squeeze(1), log_variances_prior).transpose(1, 2)
+
+        z_p = means_prior + torch.randn_like(means_prior) * torch.exp(log_variances_prior) * noise_scale
+        z = self.flow(z_p, output_padding_mask, speaker_embeddings, reverse=True)
+
+        predicted_audio = self.decoder((z * output_padding_mask), speaker_embeddings)
+        sequence_lengths = predicted_lengths * 256
```

Comment:
What is this 256?

Reply:
I think it's the upsampling rate from HiFi-GAN. I should be able to use the config for this now I think of it.